### PR TITLE
Comment module

### PR DESCRIPTION
**Pull request by hinaldobariya**
Comment-Module : Done CRUD operation 
Product-Module : Store Image of Product (WIP)

### CODE REVIEW

File: src/actions/commentAction.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    const {productId} = req.params;
+    commentModel.getByProductId(productId).then((data) => {
+      res.json({status: true, data})
+    })
+  },
+
+  delete: function (req, res) {
+    const {id} = req.params;
+    commentModel.delete(id).then(() => {
+      res.json({status: true, message: 'Comment deleted Successfully.'})
```
Comment (suresh-jarvis) [inline src/actions/commentAction.js]: `Comment deleted Successfully.` should be `Comment deleted successfully.` 
Use Lowercase for rest of message.

File: src/actions/commentAction.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    commentModel.findById(id).then((data) => {
+      res.json({status: true, data})
+    })
+  },
+
+  update: function (req, res) {
+    const {id} = req.params;
+    commentModel.update(id, req.body).then((data) => {
+      console.log(data);
+      res.json({status: true, message: 'Comment updated Successfully.', data})
```
Comment (suresh-jarvis) [inline src/actions/commentAction.js]: It's Repeat :  Same this here : use lowercase words

File: src/actions/commentAction.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    const {id} = req.params;
+    commentModel.findById(id).then((data) => {
+      res.json({status: true, data})
+    })
+  },
+
+  update: function (req, res) {
+    const {id} = req.params;
+    commentModel.update(id, req.body).then((data) => {
+      console.log(data);
```
Comment (suresh-jarvis) [inline src/actions/commentAction.js]: It's Repeat :  Remove console

File: src/actions/commentAction.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    const {id} = req.params;
+    commentModel.delete(id).then(() => {
+      res.json({status: true, message: 'Comment deleted Successfully.'})
+    })
+  },
+
+  findById: function (req, res) {
+    const {id} = req.params;
+    commentModel.findById(id).then((data) => {
+      res.json({status: true, data})
```
Comment (suresh-jarvis) [inline src/actions/commentAction.js]: It's Repeat : Check Response before send to client is found or not ?

File: src/model/comment.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+const Comment = mongoose.model('comment', CommentSchema);
+
+module.exports = {
+
+  insert: function (commentData) {
+    return new Promise((resolve, reject) => {
+      commentData['createdAt'] = moment();
+      new Comment(commentData).save(function (err, data) {
+        err == null ? resolve(data) : reject(false);
```
Comment (suresh-jarvis) [inline src/model/comment.js]: Use only resolve not reject
Same as a other query response

File: src/model/comment.js [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+      Comment.find({productId}, function (err, data) {
+        resolve((err) ? [] : (data == null) ? [] : data)
+      })
+    })
+  },
+
+  delete: function (id) {
+    return new Promise((resolve) => {
+      Comment.findOneAndRemove(id, function (err, data) {
+        resolve((err) ? [] : (data == null) ? [] : data)
```
Comment (suresh-jarvis) [inline src/model/comment.js]: In the resolve that will not return array it's boolean value

File: src/model/db.js [hunk 7]
```diff
@@ -1,5 +1,5 @@
 const mongoose = require('mongoose');
-const mongoDB = (process.env.PROD) ? 'mongodb://192.168.1.81:27017/demo' : 'mongodb://192.168.1.81:27017/demo';
+const mongoDB = (process.env.PROD) ? 'mongodb://192.168.1.82:27017/demo' : 'mongodb://192.168.1.82:27017/demo';
```
Comment (suresh-jarvis) [inline src/model/db.js]: It's Repeat : Use config.js

File: src/routes/commentRoute.js [hunk 8]
```diff
@@ -0,0 +1,12 @@
+const express = require('express');
+const router = express('Router');
+const commentAction = require('../../../product-mgmt-api/src/actions/commentAction');
```
Comment (suresh-jarvis) [inline src/routes/commentRoute.js]: This is not correct way to get file
Don't write `product-mgmt-api` in any path 
You can use : `require('../actions/commentAction');`

**Comment by hinaldobariya** [comment]
Error handling on response in comment-module

File: src/model/comment.js [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+      this.findById(id).then((resData) => {
+        if (Object.keys(resData).length > 0) {
+          Object.keys(data).map(rKey => {
+            resData[rKey] = data[rKey];
+          });
+
+          resData['updatedAt'] = moment();
+
+          resData.save(function (err, data) {
+            err ? resolve(false) : resolve(data)
```
Comment (suresh-jarvis) [inline src/model/comment.js]: Use only single resolver using ternary operator logic : Like this : `resolve((err) ? false : data)`

Comment (suresh-jarvis) [reply src/model/comment.js]: Apply all the places where you use twice resolve method

**Comment by hinaldobariya** [comment]
Resolve reviews
Adding child comment

**Pull request merged**